pkg/shell/failures.tsx

```diff
@@ -103,7 +102,7 @@ const EarlyFailureReady = ({
     );
 };
 
-export const Disconnected = ({ problem }) => {
+export const Disconnected = ({ problem } : { problem: string }) => {
```

Comment:
I dislike these sort of props as it is technical debt but just hidden for now. It is super easy to just add another thing to this and it'll become difficult to read, even right now it is a bit of a visual strain.

We can split out the props and still use a deconstructor to get problem property.
```suggestion
export interface DisconnectedProps {
    problem: string;
};

export const Disconnected = ({ problem }: DisconnectedProps) => {
```

Reply:
Yes. I am not happy with how I did this. My "standard" formatting now would be

```
export const Disconnected = ({
    problem
} : {
    problem: string
}) => {
...
}
```

(Well, hello again K&R C, long time no see! :-)

How is that?  Using an explicit name like `DisconnectedProps` (and exporting it) has the advantage that other code can derive from it, which is appropriate for a library like Patternfly, but is it maybe a bit too much for us?

I blame this all on the unfortunate fact that we can't just write

```
export const Disconnected = ({ problem: string }) => {
```

Reply:
@martinpitt, @allisonkarlitskaya, what's your opinion? Do we already have a standard for this?

Reply:
The code that I know uses the K&R style declaration. I  agree it's ugly, but good enough for me. I don't have a strong opinion about having a separate `interface DisconnectedProps` -- it feels a bit too heavy for components with only very few properties, but the duplication adds up if there are many. The cutoff point is a matter of taste/gut feeling to me, and I wouldn't personally mind if we mix these two styles.

Reply:
@martinpitt @mvollmer My leaning is that when we are dealing with props, especially object props like this, use interfaces. If they are split up like this I'm fine with inline type definitions. 

```ts
const Something = ( problem: string, name: string, extra?: string) => {
```

But for objects it is more natural for me to have interfaces. As when we are browsing code it is less to deconstruct where the props are and where the prop types are. In reality you shouldn't need to look between the two, only need to look at the interface when you are using the function.

I don't think that we need to export the interfaces either, I just suggested it if the function itself was exported. It is just a nice thing to have when you are using the function and need identical props, cause you can do:
```ts
interface Something {
    problem: string;
}

interface SomethingElse {
    string: Something["problem"]
}
```

In short though, I would love to see more usage of interfaces and more ANSI-functions than K&R style declaration of props (sorry @martinpitt!)

Reply:
I slightly prefer the interface but I'd probably skip it if there's only one property.  Changing it later is always an option>

> The type definitions look good, but consider using type instead of interface for simple object types.

This I disagree with.  The usual rule I know is "use interface when it's possible and type declaration when it's not".

Reply:
Ok, I'll introduce the *Props interfaces. Thanks!